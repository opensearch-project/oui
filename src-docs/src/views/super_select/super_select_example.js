/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from 'react';

import { renderToHtml } from '../../services';

import { GuideSectionTypes } from '../../components';

import { OuiCode, OuiSuperSelect } from '../../../../src/components';

import SuperSelectBasic from './super_select_basic';
const superSelectBasicSource = require('./super_select_basic?raw');
const superSelectBasicHtml = renderToHtml(SuperSelectBasic);
const superSelectBasicSnippet = `<OuiSuperSelect
  options={[
    {
      value: 'warning',
      inputDisplay: inputDisplay,
    },
  ]}
  valueOfSelected={value}
  onChange={onChange}
/>
`;

import SuperSelectComplex from './super_select_complex';
const superSelectComplexSource = require('./super_select_complex?raw');
const superSelectComplexHtml = renderToHtml(SuperSelectComplex);
const superSelectComplexSnippet = `<OuiSuperSelect
  options={[
    {
      value: 'option_one',
      inputDisplay: 'Option one',
      dropdownDisplay: dropdownDisplay,
    },
  ]}
  valueOfSelected={value}
  onChange={onChange}
  hasDividers
/>
`;

import SuperSelectStates from './super_select_states';
const superSelectStatesSource = require('./super_select_states?raw');
const superSelectStatesHtml = renderToHtml(SuperSelectStates);
const superSelectStatesSnippet = `<OuiSuperSelect
  options={[
    {
      value: 'option_one',
      inputDisplay: 'Option one',
    },
  ]}
  valueOfSelected={value}
  onChange={onChange}
  compressed={true}
  fullWidth={true}
  prepend={prepend}
  append={append}
/>
`;

export const SuperSelectExample = {
  title: 'Super select',
  sections: [
    {
      source: [
        {
          type: GuideSectionTypes.JS,
          code: superSelectBasicSource,
        },
        {
          type: GuideSectionTypes.HTML,
          code: superSelectBasicHtml,
        },
      ],
      text: (
        <div>
          <p>
            This is a simple replacement component for{' '}
            <strong>OuiSelect</strong> if you need more customization in either
            the display of the input or option. Simply pass an array of option
            objects:
          </p>
          <ul>
            <li>
              <OuiCode>value</OuiCode>: for storing unique value of item,{' '}
            </li>
            <li>
              <OuiCode>inputDisplay</OuiCode>: what shows inside the form input
              when selected,{' '}
            </li>
            <li>
              <OuiCode>dropdownDisplay</OuiCode>: (optional) what shows for the
              item in the dropdown
            </li>
          </ul>
          <p>
            &hellip; and the component will create a select styled button that
            triggers a popover of selectable items.
          </p>
        </div>
      ),
      props: { OuiSuperSelect },
      snippet: superSelectBasicSnippet,
      demo: <SuperSelectBasic />,
    },
    {
      title: 'More complex',
      source: [
        {
          type: GuideSectionTypes.JS,
          code: superSelectComplexSource,
        },
        {
          type: GuideSectionTypes.HTML,
          code: superSelectComplexHtml,
        },
      ],
      text: (
        <p>
          Both <OuiCode>inputDisplay</OuiCode> and{' '}
          <OuiCode>dropdownDisplay</OuiCode> accept React nodes. Therefore you
          can pass some descriptions with each option to show in the dropdown.
          If your options will most likely be multi-line, add the{' '}
          <OuiCode>hasDividers</OuiCode> prop to show borders between options.
        </p>
      ),
      props: {},
      snippet: superSelectComplexSnippet,
      demo: <SuperSelectComplex />,
    },
    {
      title: 'States',
      source: [
        {
          type: GuideSectionTypes.JS,
          code: superSelectStatesSource,
        },
        {
          type: GuideSectionTypes.HTML,
          code: superSelectStatesHtml,
        },
      ],
      text: (
        <p>
          You can pass the same props as you normally would to{' '}
          <strong>OuiSelect</strong> like disabled, isLoading, compressed,
          etc&hellip;
        </p>
      ),
      props: { OuiSuperSelect },
      snippet: superSelectStatesSnippet,
      demo: <SuperSelectStates />,
    },
  ],
};
