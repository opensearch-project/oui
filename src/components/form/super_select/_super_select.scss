/*!
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

.ouiSuperSelect__listbox {
  @include ouiScrollBar;

  max-height: 300px;
  overflow: hidden;
  overflow-y: auto;
}

.ouiSuperSelect__item {
  @include ouiFontSizeS;
  @include ouiInteractiveStates;

  padding: $ouiSizeS;
}

.ouiSuperSelect__item--hasDividers:not(:last-of-type) {
  border-bottom: $ouiBorderThin;
}
