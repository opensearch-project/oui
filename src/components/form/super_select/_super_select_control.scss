/*!
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/**
 * 1. Leave room for caret.
 * 2. Ensure the descenders don't get cut off
 * 3. Makes sure the height is correct when there's no selection
 */

.ouiSuperSelectControl {
  @include ouiFormControlStyle;
  @include ouiFormControlWithIcon($side: 'right'); /* 1 */
  @include ouiFormControlIsLoading($isNextToIcon: true);

  display: block; /* 3 */
  text-align: left;
  line-height: $ouiFormControlHeight; /* 2 */
  padding-top: 0; /* 2 */
  padding-bottom: 0; /* 2 */
  // Truncate the text
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;

  &-isInvalid {
    @include ouiFormControlInvalidStyle;
  }

  &--compressed {
    @include ouiFormControlWithIcon($side: 'right', $compressed: true); /* 1 */

    line-height: $ouiFormControlCompressedHeight; /* 2 */
    padding-top: 0; /* 2 */
    padding-bottom: 0; /* 2 */
  }

  &.ouiSuperSelect--isOpen__button { // since this is a button, we also want the visual indicator of active when options are shown
    @include ouiFormControlFocusStyle;
  }
}
