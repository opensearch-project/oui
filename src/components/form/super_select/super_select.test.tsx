/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { requiredProps } from '../../../test';

import { OuiSuperSelect } from './super_select';

jest.mock('../../portal', () => ({
  OuiPortal: ({ children }: any) => children,
}));

const options = [
  { value: '1', inputDisplay: 'Option #1' },
  { value: '2', inputDisplay: 'Option #2' },
];

describe('OuiSuperSelect', () => {
  test('is rendered', () => {
    const { container } = render(
      <OuiSuperSelect
        {...requiredProps}
        options={options}
        onChange={() => {}}
      />
    );

    expect(container).toMatchSnapshot();
  });

  describe('props', () => {
    test('fullWidth is rendered', () => {
      const { container } = render(
        <OuiSuperSelect
          {...requiredProps}
          options={options}
          onChange={() => {}}
          fullWidth
        />
      );

      expect(container).toMatchSnapshot();
    });

    test('compressed is rendered', () => {
      const { container } = render(
        <OuiSuperSelect
          {...requiredProps}
          options={options}
          onChange={() => {}}
          compressed
        />
      );

      expect(container).toMatchSnapshot();
    });

    test('is rendered with a prepend and append', () => {
      const { container } = render(
        <OuiSuperSelect
          {...requiredProps}
          options={options}
          onChange={() => {}}
          prepend="prepend"
          append="append"
        />
      );

      expect(container).toMatchSnapshot();
    });

    test('select component is rendered', () => {
      const { container } = render(
        <OuiSuperSelect
          options={[
            { value: '1', inputDisplay: 'Option #1' },
            { value: '2', inputDisplay: 'Option #2' },
          ]}
          onChange={() => {}}
        />
      );

      expect(container).toMatchSnapshot();
    });

    test('options are rendered when select is open', () => {
      const { getByTestId } = render(
        <OuiSuperSelect
          options={options}
          onChange={() => {}}
          data-test-subj="superSelect"
        />
      );

      fireEvent.click(getByTestId('superSelect'));

      expect(document.body).toMatchSnapshot();
    });

    test('valueSelected is rendered', () => {
      const { container } = render(
        <OuiSuperSelect
          options={options}
          valueOfSelected="2"
          onChange={() => {}}
        />
      );

      expect(container).toMatchSnapshot();
    });

    test('custom display is propagated to dropdown', () => {
      const { getByTestId } = render(
        <OuiSuperSelect
          options={[
            {
              value: '1',
              inputDisplay: 'Option #1',
              dropdownDisplay: 'Custom Display #1',
            },
            {
              value: '2',
              inputDisplay: 'Option #2',
              dropdownDisplay: 'Custom Display #2',
            },
          ]}
          onChange={() => {}}
          data-test-subj="superSelect"
        />
      );

      fireEvent.click(getByTestId('superSelect'));

      expect(document.body).toMatchSnapshot();
    });

    test('more props are propogated to each option', () => {
      const { getByTestId } = render(
        <OuiSuperSelect
          options={[
            { value: '1', inputDisplay: 'Option #1', disabled: true },
            {
              value: '2',
              inputDisplay: 'Option #2',
              'data-test-subj': 'option two',
            },
          ]}
          valueOfSelected="1"
          onChange={() => {}}
          data-test-subj="superSelect"
        />
      );

      fireEvent.click(getByTestId('superSelect'));

      expect(document.body).toMatchSnapshot();
    });
  });
});
