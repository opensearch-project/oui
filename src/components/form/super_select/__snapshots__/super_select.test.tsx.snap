// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OuiSuperSelect is rendered 1`] = `
<div>
  <div
    class="ouiPopover ouiPopover--anchorDownLeft ouiPopover--displayBlock ouiInputPopover ouiSuperSelect"
  >
    <div
      class="ouiPopover__anchor"
    >
      <div>
        <input
          type="hidden"
          value=""
        />
        <div
          class="ouiFormControlLayout"
        >
          <div
            class="ouiFormControlLayout__childrenWrapper"
          >
            <span
              class="ouiScreenReaderOnly"
              id="generated-id"
            >
              Select an option: , is selected
            </span>
            <button
              aria-haspopup="true"
              aria-label="aria-label"
              aria-labelledby="undefined generated-id"
              class="ouiSuperSelectControl testClass1 testClass2"
              data-test-subj="test subject string"
              type="button"
            />
            <div
              class="ouiFormControlLayoutIcons ouiFormControlLayoutIcons--right"
            >
              <span
                class="ouiFormControlLayoutCustomIcon"
              >
                <span
                  aria-hidden="true"
                  class="ouiFormControlLayoutCustomIcon__icon"
                  data-ouiicon-type="arrowDown"
                />
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`OuiSuperSelect props compressed is rendered 1`] = `
<div>
  <div
    class="ouiPopover ouiPopover--anchorDownLeft ouiPopover--displayBlock ouiInputPopover ouiSuperSelect"
  >
    <div
      class="ouiPopover__anchor"
    >
      <div>
        <input
          type="hidden"
          value=""
        />
        <div
          class="ouiFormControlLayout ouiFormControlLayout--compressed"
        >
          <div
            class="ouiFormControlLayout__childrenWrapper"
          >
            <span
              class="ouiScreenReaderOnly"
              id="generated-id"
            >
              Select an option: , is selected
            </span>
            <button
              aria-haspopup="true"
              aria-label="aria-label"
              aria-labelledby="undefined generated-id"
              class="ouiSuperSelectControl ouiSuperSelectControl--compressed testClass1 testClass2"
              data-test-subj="test subject string"
              type="button"
            />
            <div
              class="ouiFormControlLayoutIcons ouiFormControlLayoutIcons--right"
            >
              <span
                class="ouiFormControlLayoutCustomIcon"
              >
                <span
                  aria-hidden="true"
                  class="ouiFormControlLayoutCustomIcon__icon"
                  data-ouiicon-type="arrowDown"
                />
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`OuiSuperSelect props custom display is propagated to dropdown 1`] = `
<body>
  <div>
    <div
      class="ouiPopover ouiPopover--anchorDownLeft ouiPopover--displayBlock ouiInputPopover ouiSuperSelect"
    >
      <div
        aria-hidden="true"
        class="ouiPopover__anchor"
        data-aria-hidden="true"
      >
        <div>
          <input
            type="hidden"
            value=""
          />
          <div
            class="ouiFormControlLayout"
          >
            <div
              class="ouiFormControlLayout__childrenWrapper"
            >
              <span
                class="ouiScreenReaderOnly"
                id="generated-id"
              >
                Select an option: , is selected
              </span>
              <button
                aria-haspopup="true"
                aria-labelledby="undefined generated-id"
                class="ouiSuperSelectControl ouiSuperSelect--isOpen__button"
                data-test-subj="superSelect"
                type="button"
              />
              <div
                class="ouiFormControlLayoutIcons ouiFormControlLayoutIcons--right"
              >
                <span
                  class="ouiFormControlLayoutCustomIcon"
                >
                  <span
                    aria-hidden="true"
                    class="ouiFormControlLayoutCustomIcon__icon"
                    data-ouiicon-type="arrowDown"
                  />
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-hidden="true"
        data-aria-hidden="true"
        data-focus-guard="true"
        style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
        tabindex="-1"
      />
      <div
        aria-hidden="true"
        data-aria-hidden="true"
        data-focus-guard="true"
        style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
        tabindex="-1"
      />
      <div
        data-focus-lock-disabled="disabled"
      >
        <div
          aria-live="assertive"
          aria-modal="true"
          class="ouiPanel ouiPanel--borderRadiusMedium ouiPanel--plain ouiPanel--noShadow ouiPopover__panel ouiPopover__panel--bottom ouiPopover__panel-noArrow ouiPopover__panel-isAttached"
          role="dialog"
          style="top: 8px; left: 0px; will-change: transform, opacity; z-index: 2000;"
        >
          <div
            aria-hidden="true"
            class="ouiPopover__panelArrow ouiPopover__panelArrow--bottom"
            data-aria-hidden="true"
          />
          <div>
            <div
              aria-hidden="true"
              data-aria-hidden="true"
              data-focus-guard="true"
              style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
              tabindex="0"
            />
            <div
              aria-hidden="true"
              data-aria-hidden="true"
              data-focus-guard="true"
              style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
              tabindex="1"
            />
            <div
              data-focus-lock-disabled="false"
            >
              <div>
                <p
                  class="ouiScreenReaderOnly"
                  role="alert"
                >
                  You are in a form selector of 2 items and must select a single option. Use the up and down keys to navigate or escape to close.
                </p>
                <div
                  class="ouiSuperSelect__listbox"
                  role="listbox"
                  tabindex="0"
                >
                  <button
                    aria-selected="false"
                    class="ouiContextMenuItem ouiSuperSelect__item"
                    id="1"
                    role="option"
                    type="button"
                  >
                    <span
                      class="ouiContextMenu__itemLayout"
                    >
                      <span
                        class="ouiContextMenu__icon"
                        color="inherit"
                        data-ouiicon-type="empty"
                      />
                      <span
                        class="ouiContextMenuItem__text"
                      >
                        Custom Display #1
                      </span>
                    </span>
                  </button>
                  <button
                    aria-selected="false"
                    class="ouiContextMenuItem ouiSuperSelect__item"
                    id="2"
                    role="option"
                    type="button"
                  >
                    <span
                      class="ouiContextMenu__itemLayout"
                    >
                      <span
                        class="ouiContextMenu__icon"
                        color="inherit"
                        data-ouiicon-type="empty"
                      />
                      <span
                        class="ouiContextMenuItem__text"
                      >
                        Custom Display #2
                      </span>
                    </span>
                  </button>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              data-aria-hidden="true"
              data-focus-guard="true"
              style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
              tabindex="0"
            />
          </div>
        </div>
      </div>
      <div
        aria-hidden="true"
        data-aria-hidden="true"
        data-focus-guard="true"
        style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
        tabindex="-1"
      />
    </div>
  </div>
</body>
`;

exports[`OuiSuperSelect props fullWidth is rendered 1`] = `
<div>
  <div
    class="ouiPopover ouiPopover--anchorDownLeft ouiPopover--displayBlock ouiInputPopover ouiInputPopover--fullWidth ouiSuperSelect"
  >
    <div
      class="ouiPopover__anchor"
    >
      <div>
        <input
          type="hidden"
          value=""
        />
        <div
          class="ouiFormControlLayout ouiFormControlLayout--fullWidth"
        >
          <div
            class="ouiFormControlLayout__childrenWrapper"
          >
            <span
              class="ouiScreenReaderOnly"
              id="generated-id"
            >
              Select an option: , is selected
            </span>
            <button
              aria-haspopup="true"
              aria-label="aria-label"
              aria-labelledby="undefined generated-id"
              class="ouiSuperSelectControl ouiSuperSelectControl--fullWidth testClass1 testClass2"
              data-test-subj="test subject string"
              type="button"
            />
            <div
              class="ouiFormControlLayoutIcons ouiFormControlLayoutIcons--right"
            >
              <span
                class="ouiFormControlLayoutCustomIcon"
              >
                <span
                  aria-hidden="true"
                  class="ouiFormControlLayoutCustomIcon__icon"
                  data-ouiicon-type="arrowDown"
                />
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`OuiSuperSelect props is rendered with a prepend and append 1`] = `
<div>
  <div
    class="ouiPopover ouiPopover--anchorDownLeft ouiPopover--displayBlock ouiInputPopover ouiSuperSelect"
  >
    <div
      class="ouiPopover__anchor"
    >
      <div>
        <input
          type="hidden"
          value=""
        />
        <div
          class="ouiFormControlLayout ouiFormControlLayout--group"
        >
          <label
            class="ouiFormLabel ouiFormControlLayout__prepend"
          >
            prepend
          </label>
          <div
            class="ouiFormControlLayout__childrenWrapper"
          >
            <span
              class="ouiScreenReaderOnly"
              id="generated-id"
            >
              Select an option: , is selected
            </span>
            <button
              aria-haspopup="true"
              aria-label="aria-label"
              aria-labelledby="undefined generated-id"
              class="ouiSuperSelectControl ouiSuperSelectControl--inGroup testClass1 testClass2"
              data-test-subj="test subject string"
              type="button"
            />
            <div
              class="ouiFormControlLayoutIcons ouiFormControlLayoutIcons--right"
            >
              <span
                class="ouiFormControlLayoutCustomIcon"
              >
                <span
                  aria-hidden="true"
                  class="ouiFormControlLayoutCustomIcon__icon"
                  data-ouiicon-type="arrowDown"
                />
              </span>
            </div>
          </div>
          <label
            class="ouiFormLabel ouiFormControlLayout__append"
          >
            append
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`OuiSuperSelect props more props are propogated to each option 1`] = `
<body>
  <div>
    <div
      class="ouiPopover ouiPopover--anchorDownLeft ouiPopover--displayBlock ouiInputPopover ouiSuperSelect"
    >
      <div
        aria-hidden="true"
        class="ouiPopover__anchor"
        data-aria-hidden="true"
      >
        <div>
          <input
            type="hidden"
            value="1"
          />
          <div
            class="ouiFormControlLayout"
          >
            <div
              class="ouiFormControlLayout__childrenWrapper"
            >
              <span
                class="ouiScreenReaderOnly"
                id="generated-id"
              >
                Select an option: Option #1, is selected
              </span>
              <button
                aria-haspopup="true"
                aria-labelledby="undefined generated-id"
                class="ouiSuperSelectControl ouiSuperSelect--isOpen__button"
                data-test-subj="superSelect"
                type="button"
              >
                Option #1
              </button>
              <div
                class="ouiFormControlLayoutIcons ouiFormControlLayoutIcons--right"
              >
                <span
                  class="ouiFormControlLayoutCustomIcon"
                >
                  <span
                    aria-hidden="true"
                    class="ouiFormControlLayoutCustomIcon__icon"
                    data-ouiicon-type="arrowDown"
                  />
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-hidden="true"
        data-aria-hidden="true"
        data-focus-guard="true"
        style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
        tabindex="-1"
      />
      <div
        aria-hidden="true"
        data-aria-hidden="true"
        data-focus-guard="true"
        style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
        tabindex="-1"
      />
      <div
        data-focus-lock-disabled="disabled"
      >
        <div
          aria-live="assertive"
          aria-modal="true"
          class="ouiPanel ouiPanel--borderRadiusMedium ouiPanel--plain ouiPanel--noShadow ouiPopover__panel ouiPopover__panel--bottom ouiPopover__panel-noArrow ouiPopover__panel-isAttached"
          role="dialog"
          style="top: 8px; left: 0px; will-change: transform, opacity; z-index: 2000;"
        >
          <div
            aria-hidden="true"
            class="ouiPopover__panelArrow ouiPopover__panelArrow--bottom"
            data-aria-hidden="true"
          />
          <div>
            <div
              aria-hidden="true"
              data-aria-hidden="true"
              data-focus-guard="true"
              style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
              tabindex="0"
            />
            <div
              aria-hidden="true"
              data-aria-hidden="true"
              data-focus-guard="true"
              style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
              tabindex="1"
            />
            <div
              data-focus-lock-disabled="false"
            >
              <div>
                <p
                  class="ouiScreenReaderOnly"
                  role="alert"
                >
                  You are in a form selector of 2 items and must select a single option. Use the up and down keys to navigate or escape to close.
                </p>
                <div
                  aria-activedescendant="1"
                  class="ouiSuperSelect__listbox"
                  role="listbox"
                  tabindex="0"
                >
                  <button
                    aria-selected="true"
                    class="ouiContextMenuItem ouiSuperSelect__item ouiContextMenuItem-isDisabled"
                    disabled=""
                    id="1"
                    role="option"
                    type="button"
                  >
                    <span
                      class="ouiContextMenu__itemLayout"
                    >
                      <span
                        class="ouiContextMenu__icon"
                        color="inherit"
                        data-ouiicon-type="check"
                      />
                      <span
                        class="ouiContextMenuItem__text"
                      >
                        Option #1
                      </span>
                    </span>
                  </button>
                  <button
                    aria-selected="false"
                    class="ouiContextMenuItem ouiSuperSelect__item"
                    data-test-subj="option two"
                    id="2"
                    role="option"
                    type="button"
                  >
                    <span
                      class="ouiContextMenu__itemLayout"
                    >
                      <span
                        class="ouiContextMenu__icon"
                        color="inherit"
                        data-ouiicon-type="empty"
                      />
                      <span
                        class="ouiContextMenuItem__text"
                      >
                        Option #2
                      </span>
                    </span>
                  </button>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              data-aria-hidden="true"
              data-focus-guard="true"
              style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
              tabindex="0"
            />
          </div>
        </div>
      </div>
      <div
        aria-hidden="true"
        data-aria-hidden="true"
        data-focus-guard="true"
        style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
        tabindex="-1"
      />
    </div>
  </div>
</body>
`;

exports[`OuiSuperSelect props options are rendered when select is open 1`] = `
<body>
  <div>
    <div
      class="ouiPopover ouiPopover--anchorDownLeft ouiPopover--displayBlock ouiInputPopover ouiSuperSelect"
    >
      <div
        aria-hidden="true"
        class="ouiPopover__anchor"
        data-aria-hidden="true"
      >
        <div>
          <input
            type="hidden"
            value=""
          />
          <div
            class="ouiFormControlLayout"
          >
            <div
              class="ouiFormControlLayout__childrenWrapper"
            >
              <span
                class="ouiScreenReaderOnly"
                id="generated-id"
              >
                Select an option: , is selected
              </span>
              <button
                aria-haspopup="true"
                aria-labelledby="undefined generated-id"
                class="ouiSuperSelectControl ouiSuperSelect--isOpen__button"
                data-test-subj="superSelect"
                type="button"
              />
              <div
                class="ouiFormControlLayoutIcons ouiFormControlLayoutIcons--right"
              >
                <span
                  class="ouiFormControlLayoutCustomIcon"
                >
                  <span
                    aria-hidden="true"
                    class="ouiFormControlLayoutCustomIcon__icon"
                    data-ouiicon-type="arrowDown"
                  />
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-hidden="true"
        data-aria-hidden="true"
        data-focus-guard="true"
        style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
        tabindex="-1"
      />
      <div
        aria-hidden="true"
        data-aria-hidden="true"
        data-focus-guard="true"
        style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
        tabindex="-1"
      />
      <div
        data-focus-lock-disabled="disabled"
      >
        <div
          aria-live="assertive"
          aria-modal="true"
          class="ouiPanel ouiPanel--borderRadiusMedium ouiPanel--plain ouiPanel--noShadow ouiPopover__panel ouiPopover__panel--bottom ouiPopover__panel-noArrow ouiPopover__panel-isAttached"
          role="dialog"
          style="top: 8px; left: 0px; will-change: transform, opacity; z-index: 2000;"
        >
          <div
            aria-hidden="true"
            class="ouiPopover__panelArrow ouiPopover__panelArrow--bottom"
            data-aria-hidden="true"
          />
          <div>
            <div
              aria-hidden="true"
              data-aria-hidden="true"
              data-focus-guard="true"
              style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
              tabindex="0"
            />
            <div
              aria-hidden="true"
              data-aria-hidden="true"
              data-focus-guard="true"
              style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
              tabindex="1"
            />
            <div
              data-focus-lock-disabled="false"
            >
              <div>
                <p
                  class="ouiScreenReaderOnly"
                  role="alert"
                >
                  You are in a form selector of 2 items and must select a single option. Use the up and down keys to navigate or escape to close.
                </p>
                <div
                  class="ouiSuperSelect__listbox"
                  role="listbox"
                  tabindex="0"
                >
                  <button
                    aria-selected="false"
                    class="ouiContextMenuItem ouiSuperSelect__item"
                    id="1"
                    role="option"
                    type="button"
                  >
                    <span
                      class="ouiContextMenu__itemLayout"
                    >
                      <span
                        class="ouiContextMenu__icon"
                        color="inherit"
                        data-ouiicon-type="empty"
                      />
                      <span
                        class="ouiContextMenuItem__text"
                      >
                        Option #1
                      </span>
                    </span>
                  </button>
                  <button
                    aria-selected="false"
                    class="ouiContextMenuItem ouiSuperSelect__item"
                    id="2"
                    role="option"
                    type="button"
                  >
                    <span
                      class="ouiContextMenu__itemLayout"
                    >
                      <span
                        class="ouiContextMenu__icon"
                        color="inherit"
                        data-ouiicon-type="empty"
                      />
                      <span
                        class="ouiContextMenuItem__text"
                      >
                        Option #2
                      </span>
                    </span>
                  </button>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              data-aria-hidden="true"
              data-focus-guard="true"
              style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
              tabindex="0"
            />
          </div>
        </div>
      </div>
      <div
        aria-hidden="true"
        data-aria-hidden="true"
        data-focus-guard="true"
        style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
        tabindex="-1"
      />
    </div>
  </div>
</body>
`;

exports[`OuiSuperSelect props select component is rendered 1`] = `
<div>
  <div
    class="ouiPopover ouiPopover--anchorDownLeft ouiPopover--displayBlock ouiInputPopover ouiSuperSelect"
  >
    <div
      class="ouiPopover__anchor"
    >
      <div>
        <input
          type="hidden"
          value=""
        />
        <div
          class="ouiFormControlLayout"
        >
          <div
            class="ouiFormControlLayout__childrenWrapper"
          >
            <span
              class="ouiScreenReaderOnly"
              id="generated-id"
            >
              Select an option: , is selected
            </span>
            <button
              aria-haspopup="true"
              aria-labelledby="undefined generated-id"
              class="ouiSuperSelectControl"
              type="button"
            />
            <div
              class="ouiFormControlLayoutIcons ouiFormControlLayoutIcons--right"
            >
              <span
                class="ouiFormControlLayoutCustomIcon"
              >
                <span
                  aria-hidden="true"
                  class="ouiFormControlLayoutCustomIcon__icon"
                  data-ouiicon-type="arrowDown"
                />
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`OuiSuperSelect props valueSelected is rendered 1`] = `
<div>
  <div
    class="ouiPopover ouiPopover--anchorDownLeft ouiPopover--displayBlock ouiInputPopover ouiSuperSelect"
  >
    <div
      class="ouiPopover__anchor"
    >
      <div>
        <input
          type="hidden"
          value="2"
        />
        <div
          class="ouiFormControlLayout"
        >
          <div
            class="ouiFormControlLayout__childrenWrapper"
          >
            <span
              class="ouiScreenReaderOnly"
              id="generated-id"
            >
              Select an option: Option #2, is selected
            </span>
            <button
              aria-haspopup="true"
              aria-labelledby="undefined generated-id"
              class="ouiSuperSelectControl"
              type="button"
            >
              Option #2
            </button>
            <div
              class="ouiFormControlLayoutIcons ouiFormControlLayoutIcons--right"
            >
              <span
                class="ouiFormControlLayoutCustomIcon"
              >
                <span
                  aria-hidden="true"
                  class="ouiFormControlLayoutCustomIcon__icon"
                  data-ouiicon-type="arrowDown"
                />
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
